File: M_ITO_python/test_log_comparison.py
"""
Test script for log comparison - using small-scale case
"""

from iga_top2d import iga_top2d

# Use small-scale parameters for quick comparison
L = 10
W = 5  
Order = [0, 0]  # No degree elevation, keep degree 2
Num = [6, 4]    # Very small mesh
BoundCon = 1
Vmax = 0.2
penal = 3
rmin = 2

print('Running small-scale test for log comparison...')
print('Please run simultaneously in MATLAB:')
print(f'IgaTop2D({L}, {W}, [{Order[0]} {Order[1]}], [{Num[0]} {Num[1]}], {BoundCon}, {Vmax}, {penal}, {rmin})')
print('\n' + '='*80 + '\n')

X, Data, Iter_Ch = iga_top2d(L, W, Order, Num, BoundCon, Vmax, penal, rmin, case_number=999)

print('\n' + '='*80)
print('Python test completed, please compare output with MATLAB')

File: M_ITO/Test_Log_Comparison.m
% Test script for log comparison - using small-scale case
% Uses same parameters as Python version

fprintf('\nRunning small-scale test for log comparison...\n');
fprintf('Parameters: L=10, W=5, Order=[0,0], Num=[6,4], BoundCon=1, Vmax=0.2, penal=3, rmin=2\n\n');

% Call main function
IgaTop2D(10, 5, [0, 0], [6, 4], 1, 0.2, 3, 2);

fprintf('\n%s\n', repmat('=', 1, 80));
fprintf('MATLAB test completed, please compare output with Python\n');
